docs/dev-guide/deployments_table_migration.md

```diff
@@ -0,0 +1,85 @@
+# Deployments Table Migration
+
+This doc outline the data migration steps required for [PR #483 - Refactor deployments storage on redeploy and undeployment](https://github.com/caraml-dev/merlin/pull/483).
```

Comment:
CMIIW, the storage service ensure that ALL deployment are updated under the endpoint is correct during a deployment

this is to update everything outside of deployment as a one time effort

Reply:
Yes, correct. The deployment storage makes sure there will be only one running/serving deployment for a given endpoint. This migration doc is just a one-time effort to make sure the old deployment data follows the same.